hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/api/records/NodeReport.java

```diff
@@ -53,15 +53,15 @@ public static NodeReport newInstance(NodeId nodeId, NodeState nodeState,
       String httpAddress, String rackName, Resource used, Resource capability,
       int numContainers, String healthReport, long lastHealthReportTime) {
     return newInstance(nodeId, nodeState, httpAddress, rackName, used,
-        capability, numContainers, healthReport, lastHealthReportTime, null);
+        capability, numContainers, healthReport, lastHealthReportTime, null, null);
   }
 
   @Private
   @Unstable
   public static NodeReport newInstance(NodeId nodeId, NodeState nodeState,
       String httpAddress, String rackName, Resource used, Resource capability,
       int numContainers, String healthReport, long lastHealthReportTime,
-      Set<String> nodeLabels) {
+      Set<String> nodeLabels, Integer decommissioningTimeout) {
```

Comment:
Can we use int ? Am not confortable using Integer - given the sometimes arbitrary boxing/unboxing rules.

Reply:
This `Integer` comes from [`RMNode.getDecommissioningTimeout()`](https://github.com/apache/hadoop/blob/trunk/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmnode/RMNode.java#L189) that was already returning an Integer before this patch, because only nodes in DECOMMISSIONING state have an associated decommission timeout, so `null` is used to express absent timeout. In this patch `RMNode.getDecommissioningTimeout()` is used in `DefaultAMSProcessor.handleNodeUpdates` to get the argument `decommissioningTimeout` for `BuilderUtils.newNodeReport`. If we use a `int` for `decommissioningTimeout` in `NodeReport.newInstance` I think we should also use an `int` for the same argument in `BuilderUtils.newNodeReport` for uniformity, which implies a conversion from `null` to `-1` in `DefaultAMSProcessor.handleNodeUpdates`.

So I think we should either keep using `Integer decommissioningTimeout` everywhere, enconding absent timeout with `null`, or use `int decommissioningTimeout` everywhere, enconding absent timeout with a negative timeout, which is coherent with `message NodeReportProto` using an unsigned int for `decommissioning_timeout`. What do you think about these 2 alternatives?

Reply:
Yeah - sorry, just noticed that. maybe stick with Integer (although I dont like it much). Maybe we can raise another JIRA to fix it properly via your second suggestion.